rules/CVE-2021-25742/raw.rego

```diff
@@ -6,7 +6,15 @@ deny[msga] {
 	image := deployment.spec.template.spec.containers[i].image
 	is_nginx_image(image)
 	is_tag_image(image)
-	is_vulnerable(image, deployment.metadata.namespace)
+
+	# Extracting version from image tag
+	tag_version_match := regex.find_all_string_submatch_n("[0-9]+\\.[0-9]+\\.[0-9]+", image, -1)[0][0]
+    image_version_str_arr := split(tag_version_match,".")
+	image_version_arr := [to_number(image_version_str_arr[0]),to_number(image_version_str_arr[1]),to_number(image_version_str_arr[2])]
+
+	# Check if vulnerable 
+	is_vulnerable(image_version_arr, deployment.metadata.namespace)
```

Comment:
What will happen if the ingress is deployed in default namespace?

Reply:
metadata.namespace is "default" in this case